cloud/terraform/azure_config_builder_v2.py

```diff
@@ -150,19 +160,25 @@ def __new_azure_shared_image(self, instance):
             'sku': version
         }
 
+        if 'hyper_v_generation' in instance and str(instance['hyper_v_generation']) != '':
```

Comment:
FTR ` and str(instance['hyper_v_generation']) != '':` is redundant. In a boolean context the values `None` and `""` (empty string) are evaluated as `False`. The whole condition should actually be 

```
        if 'hyper_v_generation' in instance and instance['hyper_v_generation']:
```

To avoid the if-else you can also use `getattr(instance, 'hyper_v_generation', self.default_hyper_v_generation)`.